File: api-protocol/graphql/demo/src/main/java/com/example/demo/domain/reviews/repository/ReviewRepository.java
package com.example.demo.domain.reviews.repository;

import com.example.demo.domain.reviews.entity.Review;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ReviewRepository extends JpaRepository<Review, Long> {
    List<Review> findByBookIdOrderByCreatedDateDesc(Long bookId);
}

File: api-protocol/graphql/movie-dgs/src/main/kotlin/org/example/moviedgs/repositories/ReviewRepository.kt
package org.example.moviedgs.repositories

import org.example.moviedgs.entities.Movie
import org.example.moviedgs.entities.Review
import org.example.moviedgs.entities.User
import org.springframework.data.jpa.repository.JpaRepository

interface ReviewRepository : JpaRepository<Review, Long> {

    fun findAllByMovie(movie: Movie): List<Review>

    fun findAllByUser(user: User): List<Review>
}